### PR TITLE
Use `verifySeedPhrase` from `KeyringController`

### LINKED ISSUE #18762: Keyring consolidation: Convert seed phrase to `UInt8Array` when calling `verifySeedPhrase `
This is part of the effort to consolidate keyring interactions between extension and mobile.
Originally proposed [here](https://github.com/MetaMask/core/issues/1103#issuecomment-1490344485).

Since `@metamask/keyring-controller` has been introduced into the extension, we now should use `verifySeedPhrase` from its API, instead of the local implementation. 

We still need to use an array of integers internally for Background<>UI communication on the extension, so since KeyringController returns a `UInt8Array`, we need to convert it to an array of integers.

### PR DESCRIPTION
**Pull request by mikesposito**
## Explanation

This PR uses core `KeyringController`'s `verifySeedPhrase` method instead of the local implementation.

* Fixes #18762

## Manual Testing Steps

- Seed phrase reveal should work as before

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by mikesposito** [comment]
**Fitness Functions CI** is failing on this rule:

`Checking rule "Don't use `sinon` or `assert` in unit tests"`

It should be a false positive as we are not adding new test cases

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -2958,6 +2958,23 @@ export default class MetamaskController extends EventEmitter {
...
+  _uint8ArrayToBuffer(mnemonic) {
+    const recoveredIndices = Array.from(
+      new Uint16Array(new Uint8Array(mnemonic).buffer),
+    );
+    return Buffer.from(
+      new TextEncoder().encode(
+        recoveredIndices.map((i) => wordlist[i]).join(' '),
+      ),
+    );
+  }
```
[the remark this answers is not in the document]
Comment (mikesposito) [reply app/scripts/metamask-controller.js:2976]: You're right regarding the implementation, I think I got a bit confused on this, I'll use yours. Makes also sense to create a file for it, as we'll add also a function to do the other way around.

The `Uint8Array` we get from `KeyringController.verifySeedPhrase` comes from `@metamask/eth-hd-keyring`, which initialize its mnemonic with [this function](https://github.com/MetaMask/eth-hd-keyring/blob/main/src/HDKeyring.ts#L97), which always uses the english wordlist.

Apparently, we don't support multilanguage mnemonic at all - when installing a fresh extension and importing an existing seedphrase I get an error when using an Italian seedphrase.

[the remark this answers is not in the document]
Comment (mcmire) [reply app/scripts/metamask-controller.js:2976]: Sounds good.

I was reviewing a different PR and realized that "seed phrase" is our terminology but "mnemonic" is the official word from BIP-39, so we should probably be consistent:

``` suggestion
  /**
   * Decodes a BIP-39 mnemonic stored as indices of words in the English wordlist to an array of Unicode code points.
   *
   * @param {Uint8Array} wordlistIndices - Indices to specific words in the BIP-39 English wordlist.
   * @returns {Buffer} The BIP-39 mnemonic formed from the words in the English wordlist, encoded as a list of Unicode code points.
   */
  _convertWordlistIndicesToCodepoints(wordlistIndices) {
    return Array.from(
      Buffer.from(
        new TextEncoder().encode(
          wordlistIndices.map((i) => wordlist[i]).join(' '),
        ),
      ).values()
    );
  }
```

**Review by mcmire — DISMISSED** [review]
I made an edit to my suggestion earlier for `_convertWordlistIndicesToCodepoints` — basically to use "mnemonic" rather than "seed phrase". But considering that we use the name `verifySeedPhrase` in this controller, I guess we use "seed phrase" already, so the new suggestion is non-blocking. Looks good.

**Comment by mcmire** [comment]
@mikesposito Looks like there are new conflicts :(

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -2985,6 +2984,20 @@ export default class MetamaskController extends EventEmitter {
     return new Uint8Array(new Uint16Array(indices).buffer);
   }
 
+  /**
+   * Converts a BIP-39 mnemonic stored as indices of words in the English wordlist to an array of Unicode code points.
```
Comment (mcmire) [inline app/scripts/metamask-controller.js:2988]: Oops, this is inaccurate:

``` suggestion
   * Converts a BIP-39 mnemonic stored as indices of words in the English wordlist to a buffer of Unicode code points.
```

**Comment by mcmire** [comment]
@mikesposito I made a suggestion above and went ahead and applied it so that we won't have to get approvals again tomorrow in case we get both today.

**Review by mcmire — DISMISSED** [review]
Looks good!

**Comment by Gudahtt** [comment]
Whoops, more conflicts

**Review by mcmire — APPROVED** [review]
Looks good!

**Review by cryptodev-2s — APPROVED** [review]
LGTM!

**Pull request merged**